### ISSUE TITLE
BroadcastLogger#tagged without a block returns an array instead of a logger

### ISSUE THREAD

**Issue by mroach** [rails/rails#49757]
### Steps to reproduce

1. Use `BroadcastLogger#broadcast_to(logger)` to add a new logger
2. Use `BroadcastLogger#tagged("MyTag")` without a block to return a new tagged logger
3. Try to use that new tagged logger

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "rails", github: "rails/rails", branch: "main"
  gem "pry"
end

require "active_support"
require "active_support/core_ext/object/blank"
require "minitest/autorun"

class BugTest < Minitest::Test
  def test_broadcast_to_multiple_tagged_loggers
    iologger1 = Logger.new($stdout)
    iologger1 = ActiveSupport::TaggedLogging.new(iologger1)

    logger = ActiveSupport::BroadcastLogger.new(iologger1)

    # This works fine
    logger.tagged("Test1").info("OK")

    iologger2 = Logger.new($stderr)
    iologger2 = ActiveSupport::TaggedLogging.new(iologger2)

    logger.broadcast_to(iologger2)

    # This works fine
    logger.tagged("Test1") { _1.info "OK" }

    # Now we have a problem. `logger.tagged` returns an Array
    # NoMethodError: undefined method `info' for [#<Logger
    #
    # https://github.com/rails/rails/blob/ca5132b78a44677f708045905dc1b092a02f6b30/activesupport/lib/active_support/broadcast_logger.rb#L242
    logger.tagged("Test2").info("OK")

    # We'll never get here, but this fails
    assert logger.is_a?(ActiveSupport::BroadcastLogger)
  end
end

```

### Expected behavior

Using `ActiveSupport::BroadcastLogger#tagged` without a block returns a tagged logger that can be used for logging as normal.

### Actual behavior

When broadcasting to more than one logger that supports tagging, an `Array` is returned rather than an object that you can log to directly.

The specific use case where this came up: in some tests, I want to capture logs so I can test them. Previously I was replacing `Rails.logger` entirely, but now I tried adding and removing a broadcast target and I noticed this didn't work.

This appears to be caused by the `method_missing` implementation:
https://github.com/rails/rails/blob/ca5132b78a44677f708045905dc1b092a02f6b30/activesupport/lib/active_support/broadcast_logger.rb#L242

### System configuration
**Rails version**: 7.1.1

**Ruby version**: 3.2.2

**Comment by akhilgkrishnan** [comment]
looks like a similar issue of #49745. Need to investigate further more.

**Comment by davidstosik** [comment]
Similar yes. Basically `#tagged` is broken, with and without a block.
I suggest a work around in my issue, but honestly, I am not confident with it yet as I haven't been able to run our application with more than one broadcasted logger...

**Comment by Edouard-chin** [comment]
Thanks for the ping, I didn't see this issue and David's one. I have a few changes stashed that I [originally wrote it](https://github.com/rails/rails/pull/48615#issuecomment-1619209488) to fix an unrelated issue with Broadcasting + Tagged logging, and this should also fix those two issues.

Will open a PR !

**Comment by abratashov** [comment]
Yeah, I've stuck with this issue, but I thought it was related to Sentry, so the original workaround [here](https://github.com/getsentry/sentry-ruby/issues/3046)
```rb
TestJob.perform_now
# =>
TestJob
  .new(...)
  .perform(...)
```
I hope the PR for this issue will be merged 🙏
